smac/utils/multi_objective.py

```diff
@@ -1,66 +1,41 @@
-from typing import List, Optional, Tuple, Union
+from __future__ import annotations
 
 import numpy as np
 
 
-def normalize_costs(
-    values: Union[np.ndarray, List, List[List], List[np.ndarray]],
-    bounds: Optional[List[Tuple[float, float]]] = None,
-) -> np.ndarray:
-    """Normalizes the costs to be between 0 and 1 if no bounds are given. Otherwise, the costs are
-    normalized according to the bounds.
-
-    Example
-    -------
-
-    [0, 10, 5] -> [[0], [1], [0.5]]
-    [[0], [10], [5]] -> [[0], [1], [0.5]]
-    [[0, 0], [10, 50], [5, 200]] -> [[0, 0], [1, 0.25], [0.5, 1]]
+def normalize_costs(values: list[float], bounds: list[tuple[float, float]] | None = None) -> list[float]:
+    """
+    Normalizes a list of floats with corresponding bounds.
 
     Parameters
     ----------
-    values : Union[np.ndarray, List, List[List]]
-        Cost values which should be normalized.
-        If array/list is one-dimensional, it is expanded by one dimension.
-    bounds : Optional[List[Tuple[float, float]]], optional
-        Min and max bounds which should be applied to the values, by default None.
-        If bounds are None the min and max values from the data are used.
+    values : list[float]
+        List of costs to be normalized.
+    bounds : list[tuple[float, float]] | None, optional
+        List of tuple of bounds. By default None. If no bounds are passed, only ones are returned.
 
     Returns
     -------
-    np.ndarray
-        Normalized costs.
+    normalized_costs : list[float]
+        Normalized costs based on the bounds. If no bounds are given, ones are returned.
+        Also, if min and max bounds are the same, the value is set to 1.
     """
-    _values: np.ndarray
-    if isinstance(values, list):
-        _values = np.array(values)
-    else:
-        _values = values
+    if bounds is None:
```

Comment:
why does the normalize function return to kinds of things - the costs and ones (if no bounds are present) 

I would think, that if we do not have bounds then we would not normalize it, but rather return the ground truth costs instead, making this function consistent.